ironmon_tracker/screens/StatMarkingScoreSheet.lua

```diff
@@ -0,0 +1,630 @@
+					end
+				end,
+			}
+			table.insert(buttonRow.buttonList, statBtn)
+		end
+		buttonRow.score = gradeScore
+	end
+
+	-- Place button rows into the grid and update each of their contained buttons
+	SCREEN.Pager:realignButtonsToGrid(GRIDROW.X, GRIDROW.Y, 0, 0)
+
+	if SCREEN.Data.totalMarkings > 0 then
+		if SCREEN.Data.greatMarkings == SCREEN.Data.totalMarkings then
+			SCREEN.Data.percentageScore = "100"
+		else
+			local percentile = SCREEN.Data.greatMarkings * 100 / SCREEN.Data.totalMarkings
+			SCREEN.Data.percentageScore = string.format("%.1f", percentile)
+		end
+	else
+		SCREEN.Data.percentageScore = Constants.BLANKLINE
+	end
+
+	local percentile = tonumber(SCREEN.Data.percentageScore) or 0
+	SCREEN.Data.gradeLetter = SCREEN.getGradePixelImage(percentile)
+end
+
+function StatMarkingScoreSheet.checkAbilityException(statKey, abilityId)
+	if statKey == "atk" and (abilityId == AbilityData.Values.HugePowerId or abilityId == AbilityData.Values.PurePowerId) then
+		return "HugePower"
+	elseif statKey == "atk" and abilityId == AbilityData.Values.HustleId then
+		return "Hustle"
+	elseif statKey == "spd" and abilityId == AbilityData.Values.ThickFatId then
+		return "ThickFat"
+	end
+	return nil
+end
+
+---Returns true if the `statMarking` for a given `baseStat` value is within its range [and margin of error], inclusive
+---@param statMarking string
+---@param baseStat number
+---@param statKey? string Optional, pair with `abilityId` to check for abilities that would affect the marking accuracy
+---@param abilityId? number Optional
+---@return boolean
+function StatMarkingScoreSheet.isMarkingAccurate(statMarking, baseStat, statKey, abilityId)
+	local range = SCREEN.StatMarkingRanges[statMarking or false]
+	if not range then
+		return false
+	end
+	local minAllowed = range.min - SCREEN.StatMarkingRanges.MarginOfError
+	local maxAllowed = range.max + SCREEN.StatMarkingRanges.MarginOfError
+
+	local alternativeSuccess = false
+	if SCREEN.checkAbilityException(statKey, abilityId) == "HugePower" then
+		baseStat = math.min(baseStat * 2, 255) -- treat it as though it's doubled, max 255
+	elseif SCREEN.checkAbilityException(statKey, abilityId) == "Hustle" then
+		baseStat = math.min(baseStat * 1.5, 255) -- treat it as though it's boosted, max 255
+	elseif SCREEN.checkAbilityException(statKey, abilityId) == "ThickFat" then
+		local altBaseStat = math.min(baseStat * 2, 255) -- treat it as though it's doubled, max 255
+		alternativeSuccess = (altBaseStat >= minAllowed) and (altBaseStat <= maxAllowed)
+	end
+
+	return (baseStat >= minAllowed) and (baseStat <= maxAllowed) or alternativeSuccess
+end
+
+---Returns a pixel-image table that represents a letter grade from A to D.
+---@param percentile number
+---@return table pixelImage
+function StatMarkingScoreSheet.getGradePixelImage(percentile)
+	percentile = percentile or 0
+	if percentile == 100 then
+		return SCREEN.PixelImages.GRADE_A
+	elseif percentile >= 90 then
+		return SCREEN.PixelImages.GRADE_A
```

Comment:
these two if blocks are returning the same thing, any reason they're separated?

Reply:
I had originally thought I would have two pixel-images, one for "A+" and one for "A". I'll still leave this as-is, since the performance impact is nonexistent, and I may want to change the symbol in the future.